packages/ui/react-ui-menu/src/components/Toolbar/ToolbarToggleGroup.tsx

```diff
@@ -2,36 +2,44 @@
 // Copyright 2025 DXOS.org
 //
 
-import React, { useMemo } from 'react';
+import React, { useCallback } from 'react';
 
 import { Toolbar as NaturalToolbar } from '@dxos/react-ui';
 
 import type { ToolbarActionGroupProps } from './defs';
 import { type MenuAction } from '../../defs';
 import { ActionLabel } from '../ActionLabel';
+import { useMenu, useMenuItems } from '../MenuContext';
 
 const ToolbarToggleGroupItem = ({ action }: { action: MenuAction }) => {
-  const { icon, iconOnly, disabled } = action.properties;
+  const { iconSize, onAction } = useMenu();
+  const { icon, iconOnly = true, disabled } = action.properties;
+  const handleClick = useCallback(() => {
+    onAction?.(action);
+  }, [action, onAction]);
   return (
     <NaturalToolbar.ToggleGroupIconItem
       key={action.id}
       value={action.id}
       icon={icon}
+      size={iconSize}
       iconOnly={iconOnly}
       disabled={disabled}
       label={<ActionLabel action={action} />}
+      onClick={handleClick}
+      variant='ghost'
     />
   );
 };
 
-export const ToolbarToggleGroup = ({ actionGroup, graph }: ToolbarActionGroupProps) => {
-  const menuActions = useMemo(() => (graph ? (graph.actions(actionGroup) as MenuAction[]) : []), [actionGroup, graph]);
-  const { selectCardinality, value } = actionGroup.properties;
+export const ToolbarToggleGroup = ({ group, items: propsItems }: ToolbarActionGroupProps) => {
```

Comment:
if maintaining the secondary `propsItems` path is or becomes tedious you might consider only supporting the `resolveGroupItems` path which could still support static structures fine

Reply:
I did at first, but I added the `items` back as optional because it was a lot easier for consumers in certain situations e.g. NavTreeItemAction.

Reply:
The `useMenuItems` hook I think sufficiently mitigates the complexity here and clarifies to consumers a best practice for determining items.